### PR TITLE
Issue #24 - document use of existing Secrets

### PR DESCRIPTION
**Pull request by michaelwschroeder**
Signed-off-by: Mike Schroeder <mschroed@us.ibm.com>

### CODE REVIEW

File: charts/ibm-fhir-server/README.md [hunk 1]
```diff
@@ -114,6 +114,56 @@ The deployer can specify a custom search parameters named template which will be
 
 The deployer can specify custom datasource named templates which will be used in the generation of the `datasource.xml` and `bulkdata.xml` files by overriding the `datasourcesTemplate` chart value. The default for this chart value is a datasources template for a Postgres database, but this helm chart also provides named templates for Db2, Db2 on Cloud, and Derby databases in the `_datasourcesXml.tpl` file.
 
+## Using existing Secrets for sensitive data
+
+This helm chart specifies chart values for the following pieces of sensitive data:
+
+- database password and api key:
```
Comment (lmsurpre) [inline charts/ibm-fhir-server/README.md:121]:
```suggestion
- Database password or api key:
```

File: charts/ibm-fhir-server/README.md [hunk 2]
```diff
@@ -114,6 +114,56 @@ The deployer can specify a custom search parameters named template which will be
...
+    - `db.password`
+    - `db.apiKey`
+- FHIR server user and admin passwords:
+    - `fhirUserPassword`
+    - `fhirAdminPassword`
+- object storage configuration information
```
Comment (lmsurpre) [inline charts/ibm-fhir-server/README.md:127]: just for consistency
```suggestion
- Object storage configuration information:
```

File: charts/ibm-fhir-server/README.md [hunk 3]
```diff
@@ -114,6 +114,56 @@ The deployer can specify a custom search parameters named template which will be
...
+
+### Object storage configuration information
+
+To have object storage configuration information read from an existing Secret, the deployer must override the following chart values:
+
+- `objectStorage.objectStorageSecret` - this is set to the name of the Secret from which the object storage configuration information will be read
+- `objectStorage.locationSecretKey` - this is set to the key of the key/value pair within the Secret that contains the object storage location
+- `objectStorage.endpointUrlSecretKey` - this is set to the key of the key/value pair within the Secret that contains the object storage endpoint URL
+- `objectStorage.accessKeySecretKey` - this is set to the key of the key/value pair within the Secret that contains the object storage access key
+- `objectStorage.secretKeySecretKey` - this is set to the key of the key/value pair within the Secret that contains the object storage secret key
```
Comment (lmsurpre) [inline charts/ibm-fhir-server/README.md:163]: the overloaded use of `Key` really jumps out in these two.  i wish there was a different short word we could use, but I can't think of one:  both "property" and "attribute" are a bit too long.

Comment (lmsurpre) [reply charts/ibm-fhir-server/README.md:163]: `secretKeySecretKey` is especially painful :-)

File: charts/ibm-fhir-server/README.md (hunk 1 above)
Comment (michaelwschroeder) [reply charts/ibm-fhir-server/README.md]: Done

File: charts/ibm-fhir-server/README.md (hunk 2 above)
Comment (michaelwschroeder) [reply charts/ibm-fhir-server/README.md]: Done

File: charts/ibm-fhir-server/README.md (hunk 3 above)
Comment (michaelwschroeder) [reply charts/ibm-fhir-server/README.md:163]: yeah, that bothered me as well, but couldn't come up with anything better

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**